src/inference/src/dev/core_impl.cpp

```diff
@@ -711,6 +718,7 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::compile_model(const std::shared_ptr<
                                                           const ov::AnyMap& config) const {
     OV_ITT_SCOPE(FIRST_INFERENCE, ov::itt::domains::LoadTime, "Core::compile_model::model");
     std::string deviceName = device_name;
+    validDeviceName(deviceName);
```

Comment:
maybe we need to insert this function call to `parseDeviceNameIntoConfig` ? which performs all the work with device names and can perform verification as well.

Otherwise, other overloads on `compile_model`, as well as import / query model will not benefit from this function.

Reply:
I put the check in the parseDeviceNameIntoConfig and rewrote the test, since iefode reminded me that the test instance is not suitable to be placed in shared lib. Could you please help me review the PR? Thanks! @ilya-lavrenov @riverlijunjie